components/brave_shields/core/browser/brave_shields_settings_unittest.cc

```diff
@@ -0,0 +1,180 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_shields/core/browser/brave_shields_settings.h"
+
+#include "base/test/scoped_feature_list.h"
+#include "base/test/task_environment.h"
+#include "brave/components/brave_shields/core/browser/brave_shields_p3a.h"
+#include "brave/components/brave_shields/core/common/features.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/prefs/pref_service.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+using brave_shields::mojom::AdBlockMode;
+using brave_shields::mojom::FingerprintMode;
+
+class BraveShieldsSettingsTest : public testing::Test {
+ public:
+  BraveShieldsSettingsTest() {}
+  ~BraveShieldsSettingsTest() override = default;
+
+  void SetUp() override {
+    HostContentSettingsMap::RegisterProfilePrefs(profile_prefs_.registry());
+    brave_shields::RegisterShieldsP3AProfilePrefs(profile_prefs_.registry());
+    brave_shields::RegisterShieldsP3ALocalPrefs(local_state_.registry());
+    host_content_settings_map_ = new HostContentSettingsMap(
+        &profile_prefs_, false /* is_off_the_record */,
+        false /* store_last_modified */, false /* restore_session */,
+        false /* should_record_metrics */);
+    brave_shields_settings_ =
+        std::make_unique<brave_shields::BraveShieldsSettings>(
+            *host_content_settings_map_.get(), GetLocalState(),
+            &profile_prefs_);
+  }
+
+  void TearDown() override { host_content_settings_map_->ShutdownOnUIThread(); }
+
+  TestingPrefServiceSimple* GetLocalState() { return &local_state_; }
+
+  const GURL kTestUrl{"https://brave.com"};
+
+  brave_shields::BraveShieldsSettings* brave_shields_settings() {
+    return brave_shields_settings_.get();
+  }
+
+ private:
+  base::test::TaskEnvironment task_environment_;
+  TestingPrefServiceSimple local_state_;
+  sync_preferences::TestingPrefServiceSyncable profile_prefs_;
+  scoped_refptr<HostContentSettingsMap> host_content_settings_map_;
+  std::unique_ptr<brave_shields::BraveShieldsSettings> brave_shields_settings_;
+};
+
+TEST_F(BraveShieldsSettingsTest, BraveShieldsEnabled) {
+  EXPECT_TRUE(brave_shields_settings()->GetBraveShieldsEnabled(kTestUrl));
+  brave_shields_settings()->SetBraveShieldsEnabled(false, kTestUrl);
+  EXPECT_FALSE(brave_shields_settings()->GetBraveShieldsEnabled(kTestUrl));
+
+  // verify other urls unchanged
+  EXPECT_TRUE(brave_shields_settings()->GetBraveShieldsEnabled(
+      GURL("https://example.com")));
+}
+
+TEST_F(BraveShieldsSettingsTest, AdBlockMode) {
+  EXPECT_EQ(brave_shields_settings()->GetAdBlockMode(kTestUrl),
+            AdBlockMode::STANDARD);
+  brave_shields_settings()->SetAdBlockMode(AdBlockMode::AGGRESSIVE, kTestUrl);
```

Comment:
I think just a set/get here misses what we really need to test here which is that the underlying content settings for `brave_shields::GetAdControlType` and `brave_shields::GetCosmeticFilteringControlType` are set correctly. Otherwise I could the logic to return the same values, but map them to the wrong content settings or values

Reply:
Good point, I've updated the unit test to validate against the underlying content setting utility functions.
For consistency I've done the same underlying checks for each method in `BraveShieldsSettings`, to avoid similar potential issues with any method in `BraveShieldsSettings` mapping to the wrong values.